include/tools.h

```diff
@@ -225,6 +225,15 @@ IpAddress getBestPublicIps(IpMode mode);
  */
 std::string getBestPublicIp();
 
+/** Checks if IP address is available
+ *
+ * Check if the given IP address is configured on any network interface of the system
+ *
+ * @param addr the IP address to check.
+ * @return true if the IP address is available on the system.
+ */
+bool ipAvailable(const IpAddress& addr);
```

Comment:
1. This function need to be a part of the public API?
2. `IpAddress` can be a pair of addresses (one IPv4 and the other IPv6). How is this function supposed to work if both components are non-empty?

Reply:
> This function need to be a part of the public API?

I think it can be a useful function to have available

> `IpAddress` can be a pair of addresses (one IPv4 and the other IPv6). How is this function supposed to work if both components are non-empty?

Current use case was to check only one ip (in the case of the user giving one). I will refactor it to take as a parameter the mode as well and check accordingly.

Reply:
> > This function need to be a part of the public API?
> 
> I think it can be a useful function to have available

When dealing with public API that's not a good argument. Public API is a commitment. Exposing something just because it's useful may lead to unjustified maintenance overhead (like having to deal with backward compatibility issues). Let's not bloat our public API without necessity.